### PR TITLE
#20 - Add assignment as getter function of reset map

### LINKED ISSUE #20: Add getter function for assignments
Proposal: add "assignment" function as the getter function for reset maps:

```julia
assignment(hs::HybridSystem, t) = hs.resetmaps[symbol(hs, t)]
```

*Note.* If we add "ConstrainedMap" as proposed [MathematicalSystems#60](https://github.com/JuliaReach/MathematicalSystems.jl/issues/60#issue-406114195), then it is implemented as

```julia
assignment(hs::HybridSystem, t) = hs.resetmaps[symbol(hs, t)].map
```

Either way, i think it is useful to add this function because it is less implementation-dependent.

### PR DESCRIPTION
**Pull request by mforets**
Closes #20.

### CODE REVIEW

File: src/HybridSystems.jl [hunk 1]
```diff
@@ -104,6 +104,13 @@ guard(hs::HybridSystem, t) = stateset(hs.resetmaps[symbol(hs, t)])
 # because guards are given as the state constraints of the reset map
 MathematicalSystems.stateset(hs::HybridSystem, t) = guard(hs, t)
 
+"""
+    assignment(hs::HybridSystem, t)
+
+Returns the assignment for the transition `t`.
+"""
+assignment(hs::HybridSystem, t) = hs.resetmaps[symbol(hs, t)]
```
Comment (blegat) [inline src/HybridSystems.jl]: `guard` could use `assignment` in his implementation

**Comment by schillic** [comment]
I find this implementation misleading. In the standard terminology, the assignment is the part of the transition that is applied after taking the transition. In this package, `resetmaps` wraps both the (usual) assignment and the guard.

**Comment by blegat** [comment]
That's a good point, what do you suggest ?

**Comment by mforets** [comment]
I see what you mean; but the "problem" is that the maps types, not functions. (we could otherwise let `assignment` return the `apply` function ... ).

Consider [here](https://people.eecs.berkeley.edu/~sseshia/219c/spr16/lectures/Frehse_SyDe15.pdf) (which refers to [Henzinger's](https://pub.ist.ac.at/~tah/Publications/the_theory_of_hybrid_automata.pdf)), it calls it "jump relation" `Jump(e)` as `(guard, assignment)` tuples for each edge.

**Comment by schillic** [comment]
Yes, I would change the name. "Jump relation" could also be misunderstood as the transition relation. `resetmap` would be in the terminology of this package.

**Comment by mforets** [comment]
Just that? To let `resetmap(hs, t) = hs.resetmaps[symbol(hs, t)]` ?  And then we can do `apply(resetmap(hs, t), X)`

And how about `assignment(hs, t) = x -> apply(hs.resetmaps[symbol(hs, t)], x...)`?

**Comment by schillic** [comment]
Hm, thinking about this, I kind of disagree again. `guard` is a getter, while `assignment` is proposed to apply the assignment. How about `apply_assignment` instead?

**Comment by mforets** [comment]
`assignment` doesn't apply the assignment; it returns the function. So i understand it also as a getter. It is different from `assignment(hs, t, args)`.

**Comment by schillic** [comment]
You wrote `assignment(hs, t) = x -> apply(hs.resetmaps[symbol(hs, t)], x...)` above.
Are you saying that `apply` does not apply? :scream_cat:

**Pull request merged**